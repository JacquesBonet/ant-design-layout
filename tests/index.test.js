import pkg from '../package.json';

const testDist = process.env.LIB_DIR === 'dist';

describe('ant-design-layout dist files', () => {
  it('exports modules correctly', () => {
    // eslint-disable-next-line global-require,import/no-unresolved
    const antd = testDist ? require('../dist/ant-design-layout') : require('../components');
    expect(Object.keys(antd)).toMatchSnapshot();
  });

  // https://github.com/ant-design/ant-design/issues/1970
  // https://github.com/ant-design/ant-design/issues/1804
  if (testDist) {
    it('ant design layout should export version', () => {
      // eslint-disable-next-line global-require,import/no-unresolved
      const antd = require('../dist/ant-design-layout');
      expect(antd).toBeTruthy();
      expect(antd.version).toBe(pkg.version);
    });

    it('ant-design-layout should export version', () => {
      // eslint-disable-next-line global-require,import/no-unresolved
      const antd = require('../dist/ant-design-layout.min');
      expect(antd).toBeTruthy();
      expect(antd.version).toBe(pkg.version);
    });
  }
});
