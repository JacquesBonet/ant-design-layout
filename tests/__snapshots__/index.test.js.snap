// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ant-design-layout dist files exports modules correctly 1`] = `
Array [
  "Col",
  "Grid",
  "Row",
]
`;
